### ISSUE TITLE
Support for ssl root ca certificate store information

### ISSUE THREAD

**Issue by Chaz6** [facebook/osquery#2553]
Please consider implementing support for providing information about the installed root CA certificates.

Here are my suggestions for a name and fields:-

Table name: ca_certificates
Fields: filename, fingerprint, fingerprint_algorithm,trust_websites, trust_emails, trust_code, name, valid_from, valid_to, serial_number, subject, friendly_name, issuer, certificate_purposes, extended_validation_policy_oids, ocsp_urls,

**Comment by theopolis** [comment]
Hi @Chaz6! Were you interested in this information for Windows hosts? We have a `certificates` table on OS X: https://osquery.io/docs/tables/#certificates with the following schema:
```python
table_name("certificates")
description("Certificate Authorities installed in Keychains/ca-bundles.")
schema([
    Column("common_name", TEXT, "Certificate CommonName"),
    Column("subject", TEXT, "Certificate distinguished name"),
    Column("issuer", TEXT, "Certificate issuer distinguished name"),
    Column("ca", INTEGER, "1 if CA: true (certificate is an authority) else 0"),
    Column("self_signed", INTEGER, "1 if self-signed, else 0"),
    Column("not_valid_before", DATETIME, "Lower bound of valid date"),
    Column("not_valid_after", DATETIME, "Certificate expiration data"),
    Column("signing_algorithm", TEXT, "Signing algorithm used"),
    Column("key_algorithm", TEXT, "Key algorithm used"),
    Column("key_strength", TEXT, "Key size used for RSA/DSA, or curve name"),
    Column("key_usage", TEXT, "Certificate key usage and extended key usage"),
    Column("subject_key_id", TEXT, "SKID an optionally included SHA1"),
    Column("authority_key_id", TEXT, "AKID an optionally included SHA1"),
    Column("sha1", TEXT, "SHA1 hash of the raw certificate contents"),
    Column("path", TEXT, "Path to Keychain or PEM bundle"),

])
attributes(cacheable=True)
```

That seems to capture everything by the OIDs in the extended policy/configuration and OCSP information. It will be quite a bit of work to replicate the table, but that's definitely in the road map. Let us know if you'd be interested in writing some C++!

**Comment by juju4** [comment]
Same discussion than for Windows (#2922). It would be great to have.
One limitation/extra work would be to include firefox user profile certificate (cert8.db) as firefox is using its own db.

**Comment by fmanco** [comment]
If I understand correctly this request was for Windows which landed #3498 so I'm going to close. Feel free to reopen if needed with more details.

**Comment by Chaz6** [comment]
I would also like this for Linux, however there is currently no standard location. I suggest that the XDG Base Directory specification should be updated to include a path to system root certificates.
